bids-validator/src/schema/applyRules.ts

```diff
@@ -291,12 +294,12 @@ export function evalColumns(
       if (
         !typeCheck(value)
       ) {
-        context.dataset.issues.addNonSchemaIssue(error_code, [
-          {
-            ...context.file,
-            evidence: `'${value}' ${inspect(columnObject)}`,
-          },
-        ])
+        context.dataset.issues.add({
+          code: error_code,
+          location: context.path,
+          issueMessage: `'${value}' ${inspect(columnObject)}`,
```

Comment:
```suggestion
          code: error_code,
          subCode: name
          location: context.path,
          issueMessage: `'${value}'`,
```

We could probably add row number if we enumerate the loop.